src/commands/selectCommandTemplate.ts

```diff
@@ -159,3 +184,47 @@ async function quickPickTemplate(context: IActionContext, templates: CommandTemp
 
     return selection.data;
 }
+
+function getConstrainedTemplates(templates: CommandTemplate[], matchContext: string[], currentContextType: ContextType): CommandTemplate[] {
+    return templates.filter(template => {
+        if (!template.contextTypes && !template.match) {
+            // If neither contextTypes nor match is defined, this is an unconstrained template

```

Comment:
Add `isConstrainedTemplate` property to `CommandTemplate`? You could use it here and in `getUnconstrainedTemplates` method below

Reply:
I'm not sure I follow. If I added such a property I would basically have to apply the same logic in these methods in order to set it; wouldn't it be about the same amount of code?

Reply:
(If `CommandTemplate` was a class I could easily use a `get` method but it's just an interface...)

Reply:
OK never mind then